In [2]:
# Import ..
import pandas as pd
import seaborn as sns
from matplotlib import pyplot as plt



### 1. Load data and get an overview

In [3]:
path = r"C:\Users\ThinkPad\Desktop\Mastermind\WHO\WHO_health_indicators\WHS2025_DATADOWNLOAD.csv"
df = pd.read_csv(path)
df.info()


<class 'pandas.DataFrame'>
RangeIndex: 9729 entries, 0 to 9728
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   IndicatorName   9729 non-null   str    
 1   IndicatorCode   9729 non-null   str    
 2   Location        9729 non-null   str    
 3   LocationCode    9729 non-null   str    
 4   Year            9729 non-null   str    
 5   Disaggregation  1517 non-null   str    
 6   NumericValue    9665 non-null   float64
 7   DisplayValue    9727 non-null   str    
 8   Comments        2722 non-null   str    
dtypes: float64(1), str(8)
memory usage: 684.2 KB


In [4]:
df.columns

Index(['IndicatorName', 'IndicatorCode', 'Location', 'LocationCode', 'Year',
       'Disaggregation', 'NumericValue', 'DisplayValue', 'Comments'],
      dtype='str')

### 2. Check if indicator name is connected to indicator code

In [ ]:


# 1. Check if the mapping is one-to-one (each name maps to exactly one code, and vice versa)
name_to_code = df.groupby('IndicatorName')['IndicatorCode'].nunique()
code_to_name = df.groupby('IndicatorCode')['IndicatorName'].nunique()

print("Names with multiple codes:", (name_to_code > 1).sum())
print("Codes with multiple names:", (code_to_name > 1).sum())




Names with multiple codes: 3
Codes with multiple names: 2


### -> Strong connection but exception

### 3. Focussion on everything with alcohol in the Name

In [6]:
# Looking at alcohol

## get the indicator names

df_alc = df[(df["IndicatorName"]).str.contains("alcohol", case=False)]
df_alc.info()

<class 'pandas.DataFrame'>
Index: 201 entries, 1247 to 8975
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   IndicatorName   201 non-null    str    
 1   IndicatorCode   201 non-null    str    
 2   Location        201 non-null    str    
 3   LocationCode    201 non-null    str    
 4   Year            201 non-null    str    
 5   Disaggregation  6 non-null      str    
 6   NumericValue    201 non-null    float64
 7   DisplayValue    201 non-null    str    
 8   Comments        0 non-null      str    
dtypes: float64(1), str(8)
memory usage: 15.7 KB


In [27]:
df_alc.head(10)

,IndicatorName,IndicatorCode,Location,LocationCode,Year,Disaggregation,NumericValue,DisplayValue,Comments
1247,Contact coverage of treatment services for alc...,SUD_TREATMENTSERVICES_COVERAGE,Angola,AGO,2016,ALCOHOL,0.406217,0.4,NaN
1248,Contact coverage of treatment services for alc...,SUD_TREATMENTSERVICES_COVERAGE,Barbados,BRB,2016,ALCOHOL,0.266751,0.3,NaN
1249,Contact coverage of treatment services for alc...,SUD_TREATMENTSERVICES_COVERAGE,Belgium,BEL,2018,ALCOHOL,1.167881,1.2,NaN
1250,Contact coverage of treatment services for alc...,SUD_TREATMENTSERVICES_COVERAGE,Morocco,MAR,2016,ALCOHOL,7.082153,7.1,NaN
1251,Contact coverage of treatment services for alc...,SUD_TREATMENTSERVICES_COVERAGE,Netherlands (Kingdom of the),NLD,2016,ALCOHOL,13.824685,13.8,NaN
1252,Contact coverage of treatment services for alc...,SUD_TREATMENTSERVICES_COVERAGE,Türkiye,TUR,2016,ALCOHOL,1.413435,1.4,NaN
8781,Total alcohol per capita (≥ 15 years of age) c...,SA_0000001688,Afghanistan,AFG,2022,NaN,0.007942,0.0,NaN
8782,Total alcohol per capita (≥ 15 years of age) c...,SA_0000001688,African Region,AFR,2022,NaN,3.462375,3.5,NaN
8783,Total alcohol per capita (≥ 15 years of age) c...,SA_0000001688,Albania,ALB,2022,NaN,4.974771,5.0,NaN
8784,Total alcohol per capita (≥ 15 years of age) c...,SA_0000001688,Algeria,DZA,2022,NaN,0.455062,0.5,NaN


### 4. Gain information for creating a new df about alcohol
4.1 How many different values are there in each column and how often do the repeat?

4.2 If there is value A and B in colum 1 and value C and D in column 2: is A always connected with C and B always with D?

In [ ]:
# 4.1
# print(df_alc["IndicatorCode"].unique())
# print(df_alc["IndicatorName"].unique())
print(df_alc["IndicatorName"].value_counts())
print(df_alc["IndicatorCode"].value_counts())




IndicatorName
Total alcohol per capita (≥ 15 years of age) consumption (litres of pure alcohol)    195
Contact coverage of treatment services for alcohol use disorders (%)                   6
Name: count, dtype: int64
IndicatorCode
SA_0000001688                     195
SUD_TREATMENTSERVICES_COVERAGE      6
Name: count, dtype: int64


In [19]:
#4.2
df_alc.groupby("IndicatorName")["IndicatorCode"].unique()

IndicatorName
Contact coverage of treatment services for alcohol use disorders (%)                 [SUD_TREATMENTSERVICES_COVERAGE]
Total alcohol per capita (≥ 15 years of age) consumption (litres of pure alcohol)                     [SA_0000001688]
Name: IndicatorCode, dtype: object

### -> There are two different variables in column IndicatorName and 2 different variables in column B.
- The variables are connected with each other
- only 6 entrys for Treatment coverage but 195 for total alc.

### 5. Create two new dataframes: one for each category (even though one will only have 6 observations)


In [21]:
df_consum_alc = df[df["IndicatorCode"]=="SA_0000001688"]
df_treatment_alc = df[df["IndicatorCode"]=="SUD_TREATMENTSERVICES_COVERAGE"]
display(df_consum_alc.info())
display(df_treatment_alc.info())

<class 'pandas.DataFrame'>
RangeIndex: 195 entries, 8781 to 8975
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   IndicatorName   195 non-null    str    
 1   IndicatorCode   195 non-null    str    
 2   Location        195 non-null    str    
 3   LocationCode    195 non-null    str    
 4   Year            195 non-null    str    
 5   Disaggregation  0 non-null      str    
 6   NumericValue    195 non-null    float64
 7   DisplayValue    195 non-null    str    
 8   Comments        0 non-null      str    
dtypes: float64(1), str(8)
memory usage: 13.8 KB


None

<class 'pandas.DataFrame'>
RangeIndex: 22 entries, 1247 to 1268
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   IndicatorName   22 non-null     str    
 1   IndicatorCode   22 non-null     str    
 2   Location        22 non-null     str    
 3   LocationCode    22 non-null     str    
 4   Year            22 non-null     str    
 5   Disaggregation  22 non-null     str    
 6   NumericValue    22 non-null     float64
 7   DisplayValue    22 non-null     str    
 8   Comments        0 non-null      str    
dtypes: float64(1), str(8)
memory usage: 1.7 KB


None

In [23]:
# how many different years?
df_consum_alc["Year"].value_counts()

Year
2022    195
Name: count, dtype: int64

# -> As the df only contains the year 2022 for the corresponding interest (alcohol), downloaded a new df from WHO, only about alcohol